src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        Node<K, V> newNode = new Node<>(key, value);
+        if (table[getBucket(newNode)] == null) {
+            table[getBucket(newNode)] = newNode;
+        } else if (ifEqualsKey(table[getBucket(newNode)], key)) {
+            table[getBucket(newNode)].value = value;
+            return;
+        } else {
+            Node<K, V> currentNode = table[getBucket(newNode)];
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+                if (ifEqualsKey(currentNode, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+            }
+            currentNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> currentNode : table) {
```

Comment:
we can get value by index, and after that, we can use while loop and find the node in the list

Reply:
can you please explain why should we complicate code in this case and add search by index